src/lib/draw-and-pick.js

```diff
@@ -136,16 +136,21 @@ export function pickLayers(gl, {
 
     affectedLayers.forEach(layer => {
       let info = Object.assign({}, baseInfo);
-      info.layer = layer;
 
       if (layer === pickedLayer) {
         info.color = pickedColor;
         info.index = pickedObjectIndex;
         info.picked = true;
       }
 
-      // Let layers populate its own info object
-      info = layer.pickLayer({info, mode});
+      // walk up the composite chain and find the owner of the event
+      // sublayers are never directly exposed to the user
+      while (layer && info) {
+        info.layer = layer;
+        // Let layers populate its own info object
+        info = layer.pickLayer({info, mode});
```

Comment:
Chaining is needed here because the sublayer is the one that can interpret the encoded picking color, and the composite layer is the one that knows the original user data.
We definitely should have a dedicated doc about picking. I can take a stab at it once the demo site is fixed.